### PR TITLE
MINOR : Refactor service principal configuration

### PR DESCRIPTION
**Pull request by raju-saravanan**

### CODE REVIEW

File: webservice/src/main/java/com/hortonworks/registries/webservice/RegistryApplication.java [hunk 1]
```diff
@@ -94,23 +94,24 @@ public void run(RegistryConfiguration registryConfiguration, Environment environ
     }
 
     private void initializeUGI(RegistryConfiguration conf) throws IOException {
-        LOG.debug("Initialization of User Group ininformation...");
+        LOG.debug("Initialization of User Group information...");
         if (UserGroupInformation.isSecurityEnabled()) {
```
Comment (guruchai) [inline webservice/src/main/java/com/hortonworks/registries/webservice/RegistryApplication.java]: This check i believe requires core-site.xml. Can we not use the conf.getServiceAuthenticationConfiguration() itself?

File: webservice/src/main/java/com/hortonworks/registries/webservice/RegistryApplication.java [hunk 2]
```diff
@@ -94,23 +94,24 @@ public void run(RegistryConfiguration registryConfiguration, Environment environ
...
             LOG.debug("UGI.isSecurityEnabled() = true.");
 
-            AuthMethodConfiguration authMethodConf = conf.getAuthenticationMethod();
+            ServiceAuthenticationConfiguration authMethodConf = conf.getServiceAuthenticationConfiguration();
             if(authMethodConf != null) {
-                String serverPrincipal = authMethodConf.getServerPrinciple();
-                String keyTab = authMethodConf.getServerPrincipleKeytab();
+                Map<String,String> serviceAuthenticationProperties = authMethodConf.getProperties();
+                String principal = serviceAuthenticationProperties.get("principal");
+                String keytab = serviceAuthenticationProperties.get("keytab");
```
Comment (guruchai) [inline webservice/src/main/java/com/hortonworks/registries/webservice/RegistryApplication.java]: You should validate the principal and keytab for null and empty.

You should also catch exception from below, log it as error and then fallback to simple.

> UserGroupInformation.loginUserFromKeytab(principal, keytab);

**Comment by raju-saravanan** [comment]
@guruchai: Addressed your review comments, can you have another look?

File: webservice/src/main/java/com/hortonworks/registries/webservice/RegistryApplication.java [hunk 3]
```diff
@@ -94,29 +95,33 @@ public void run(RegistryConfiguration registryConfiguration, Environment environ
...
+                    String principal = serviceAuthenticationProperties.get("principal");
+                    String keytab = serviceAuthenticationProperties.get("keytab");
+
+                    if (StringUtils.isNotEmpty(principal) && StringUtils.isNotEmpty(keytab)) {
+                        LOG.debug("Login with principal = '" + principal + "' and keyTab = '" + keytab + "'");
+                        try {
+                            UserGroupInformation.loginUserFromKeytab(principal, keytab);
+                            LOG.debug("Successfully logged in");
+                        } catch (Exception e) {
+                            LOG.error("Failed to log in");
```
Comment (guruchai) [inline webservice/src/main/java/com/hortonworks/registries/webservice/RegistryApplication.java]: Please log the exception.

**Comment by raju-saravanan** [comment]
Thanks @guruchai , addressed your comments.

**Pull request merged**